features/FEATURE_BLE/targets/TARGET_CORDIO/source/CordioBLE.cpp

```diff
@@ -273,8 +291,10 @@ void BLE::processEvents()
                 DmExtConnSlaveInit();
             }
 
+#if BLE_FEATURE_GATT_SERVER
             deviceInstance().getGattServer().initialize();
             deviceInstance().initialization_status = INITIALIZED;
+#endif // BLE_FEATURE_GATT_SERVER
```

Comment:
I think `deviceInstance().initialization_status = INITIALIZED;` is independent of `BLE_FEATURE_GATT_SERVER`.